torch/nn/functional.py

```diff
@@ -341,7 +341,7 @@ def max_pool1d(input, kernel_size, stride=None, padding=0, dilation=1,
 
     See :class:`~torch.nn.MaxPool1d` for details.
     """
-    ret = torch.max_pool1d(input, kernel_size, stride, padding, dilation, ceil_mode)
+    ret = torch.max_pool1d_with_indices(input, kernel_size, stride, padding, dilation, ceil_mode)
```

Comment:
I just realized that these functions do not respect `return_indices` anymore. Can you fix them? 

Could you also add a test for this option in `test_nn.py`?

Reply:
Why? Doesn't `return ret if return_indices else ret[0]` do it?

Reply:
Oh sorry you are right. Missed the line below somehow.